src/group_theory/finite_abelian.lean

```diff
@@ -17,9 +17,29 @@ import algebra.module.pid
 
 -/
 
-
 open_locale direct_sum
 
+namespace module
+
+open_locale classical
+
+/-- A finitely generated torsion `ℤ`-module is finite. -/
+noncomputable def fintype_of_fg_torsion (G : Type*) [add_comm_group G] [module ℤ G]
```

Comment:
This is true more generally for... Dedekind domains?

Reply:
Is it true? I thought for PIDs like C[X] we can have modules like C[X]/<X> which are finitely generated torsion but infinite.

Reply:
Oh, my bad! Can you generalize it to rings that are finite over `ℤ` if it not too much trouble? I have in mind ring of integers of number fields.

Reply:
I think it's going to be some effort, as we basically have to rewrite this using the classification `submodule.is_internal_prime_power_torsion` in `algebra/module/dedekind_domain`, and the correct generalisation for this are called "residually finite" rings I believe. What application do you have in mind that needs finiteness of modules specifically over rings of integers?

Reply:
Ok, let's keep it for `Z`.